### PR TITLE
[485573]: New Vector operations added

### PR DESCRIPTION
**Pull request by mcaniga**
Done

### CODE REVIEW

**Comment by mcaniga** [comment]
-- pr1 (commits are sorted from newest to oldest) (new commits)
a7596fc3f0e7bfe29ebafdc728dfae301b7c32ad - Readme updated with new Vector operations
24bf74209a12ed613d479069a0d1465f3e6d1ac4 - Travis build badge
d94cdc5780ca1153e4cebf81716fdd926c2d1086 - Travis build fixed
473244e40d13193068a9e84c6cdbd1d40e64c055 - Tests for added Vector methods
65d7a2e8ab0ba09adf95ee7d301970d4c08480d5 - Implemented Vector methods marked as TODO

--pr1 after squash (new commits)
50679d48af8d07959d7f10820db16179a006277d - Implemented Vector methods marked as TODO

--pr2 (new commits)
d163e24a534dba05018eab3c7806c7f4a8744657 - Remaining Vector methods added
2de8b5428b2c25ee6001b7875c4dd0367de0cc53 - Tests for Vector __and__ method
ec0f0b6bcabf87b91f9221a322c4890a3f7aa540 - Unit tests fixed

--pr2 after squash (new commits)
546802cd3d16e8820f85c82982ea9d73bb4e4350 - Implemented remaining Vector methods marked as TODO

**Comment by ph4r05** [comment]
Travis passes but 0 tests is executed, pls fix this.

File: .travis.yml [hunk 1]
```diff
@@ -10,7 +10,7 @@ addons:
 
 python:
   - "3.6"
-  - "3.7"
+#  - "3.7"
```
Comment (ph4r05) [inline .travis.yml]: You could have added tests for 3.7, I mean, fix why 3.7 is not working not commenting it out :)

**Review by ph4r05 — CHANGES_REQUESTED** [review]
Pls fix the Travis, 0 tests executed

**Comment by ph4r05** [comment]
@cucak559 pls squash the commits to one single signed commit. then I can approve and award you HW points. Pls tag me in comment here when you are done. Thx

**Comment by mcaniga** [comment]
@ph4r05 Commits squashed

**Review by ph4r05 — APPROVED** [review]

**Pull request closed without merging**